plugins/@grouparoo/mongo/src/lib/table-import/getChangedRows.ts

```diff
@@ -12,21 +12,28 @@ export const getChangedRows: GetChangedRowsMethod = async ({
   sourceOffset,
   highWaterMarkAndSortColumnASC,
   secondarySortColumnASC,
+  matchConditions,
   highWaterMarkKey,
 }) => {
-  let query = {};
   let aggPipeline = [];
+
   if (highWaterMarkCondition) {
     highWaterMarkCondition.value = new Date(
       String(highWaterMarkCondition.value)
     );
-    query = makeFindQuery(highWaterMarkCondition);
+    const query = makeFindQuery(highWaterMarkCondition);
+    if (Object.keys(query).length > 0) {
+      aggPipeline.push({ $match: query });
+    }
   }
-  if (Object.keys(query).length > 0) {
-    aggPipeline.push({
-      $match: query,
-    });
+
+  for (const condition of matchConditions) {
+    const filterQuery = makeFindQuery(condition);
+    if (Object.keys(filterQuery).length > 0) {
+      aggPipeline.push({ $match: filterQuery });
+    }
```

Comment:
@pauloouriques This is working just fine - here's the query generated:

```json
[
   {
      "$match":{
         "id":{
            "$gt":4
         }
      }
   },
   {
      "$match":{
         "id":{
            "$lt":7
         }
      }
   },
   {
      "$sort":{
         "stamp":1,
         "id":1
      }
   },
   {
      "$addFields":{
         "__hwm":"$stamp"
      }
   },
   {
      "$limit":100
   },
   {
      "$skip":0
   }
]
```

Is there any perfomance penalty for having 2 separate comparison blocks for the "id" property, rather than something like:

```json 
   {
      "$match":{
         "id":[{ "$gt":4 }, { "$lt":7 }]
      }
   },
```